esp/scm/ws_topology.ecm

```diff
@@ -415,14 +415,10 @@ ESPresponse [exceptions_inline,encode(0)] TpClusterInfoResponse
 };
 
 
-ESPenum TpMachineType : int
-{
-    TPMACHINECLUSTER(0, "TPMACHINECLUSTER")
-};
 ESPrequest [] TpMachineQueryRequest
 {
     //Type of Node the parent is....
-    TpMachineType Type;
+    string        Type;
```

Comment:
The string works. But, I am wondering whether it should be a string ESPenum. Like:
ESPenum TpMachineType : string
{
        THORMACHINES("THORMACHINES"),
        ROXIEMACHINES("ROXIEMACHINES"),
        ...
};
For backward compatible, you may version out the '[depr_ver("?")]  string        Type;' and add new '[min_ver("?")] ESPenum TpMachineType MachineType;'.

Reply:
Looking at the method void CTpWrapper::getClusterMachineList, the second parameter is the Type field. That method compares to a considerable number of different values that are not defined, but are literal string compares. I believe changing to a string enum would require additional code changes to ensure all usages of _Type_ used values defined by the enum.

Reply:
@kenrowland has added https://track.hpccsystems.com/browse/HPCC-31458.